src/System.Net.Http/src/uap/System/Net/HttpHandlerToFilter.cs

```diff
@@ -47,9 +50,27 @@ internal HttpHandlerToFilter(RTHttpBaseProtocolFilter filter)
             }
             cancel.ThrowIfCancellationRequested();
 
-
             RTHttpRequestMessage rtRequest = await ConvertRequestAsync(request).ConfigureAwait(false);
-            RTHttpResponseMessage rtResponse = await _next.SendRequestAsync(rtRequest).AsTask(cancel).ConfigureAwait(false);
+
+            RTHttpResponseMessage rtResponse;
+            try
+            {
+                rtResponse = await _next.SendRequestAsync(rtRequest).AsTask(cancel).ConfigureAwait(false);
+            }
+            catch (TaskCanceledException)
+            {
+                throw;
+            }
+            catch (Exception)
```

Comment:
First, I changed the "savedException" of type `Exception` to be a "savedExceptionDispatchInfo" of type `ExceptionDispatchInfo` as recommended by @stephentoub.

If there is a saved ExceptionDispatchInfo, we will use it because at the point it happened, the user callback threw an unhandled exception.  We need to report that. It also stops all further processing of the HTTP request, so there are no other exceptions we need to worry about.

When such an exception is thrown, our WinRTCertificateCallback wrapper then calls Reject() on the WinRT callback layer. That causes the WinRT layer to throw an exception.  That is really the exception we are trapping here in that case. And so, we need to see if we saved the ExceptionDispatchInfo object because that is really the root cause of why the WinRT layer threw an exception. And so, we need to propagate that.